Lib/smtplib.py

```diff
@@ -662,7 +666,7 @@ def auth_plain(self, challenge=None):
     def auth_login(self, challenge=None):
         """ Authobject to use with LOGIN authentication. Requires self.user and
         self.password to be set."""
-        if challenge is None:
+        if challenge is None or self._auth_challenge_count < 2:
```

Comment:
The [draft standard for AUTH LOGIN](https://tools.ietf.org/html/draft-murchison-sasl-login-00) specifies that [clients SHOULD ignore](https://tools.ietf.org/html/draft-murchison-sasl-login-00#section-2.1) the actual contents of the challenge, and responded only based on how many challenges have been received.